modules/_labs/select/react/lib/SelectMenu.tsx

```diff
@@ -133,20 +189,152 @@ const MenuList = styled('ul')<Pick<SelectProps, 'error' | 'theme'>>(
+    {
+      name: 'flip',
+      options: {
+        fallbackPlacements,
+      },
+    },
+    {
+      name: 'offset',
+      options: {
+        offset: ({
+          placement,
+          reference,
+          popper,
+        }: {
+          placement: Placement;
+          reference: Rect;
+          popper: Rect;
+        }) => {
+          // Skid menu along the edge of the button to account
+          // for fractional x-positioning of the button (e.g.,
+          // if the button is in a horizontally-centered modal)
+          // and to ensure proper alignment between the button
+          // and the menu
+          const skidding = reference.x % 1 >= 0.5 ? 1 : 0;
+
+          // Displace menu towards the button to obscure the bottom
+          // edge of the button and to create a smooth visual
+          // connection between the button and the menu
+          const distance = -parseInt(borderRadius.m, 10);
+
+          return [skidding, distance];
+        },
+      },
+    },
+  ];
+
+  return {
+    modifiers,
+    onFirstUpdate: () => {
+      if (isAutoFocused && menuRef && menuRef.current) {
+        menuRef.current.focus();
+      }
+    },
+  };
+};
+
 const SelectMenu = (props: SelectMenuProps) => {
-  const {children, error, isAnimated, isHiding, menuRef, ...elemProps} = props;
-
-  return (
-    <Menu error={error} isAnimated={isAnimated} isHiding={isHiding}>
-      <MenuList error={error} ref={menuRef} role="listbox" tabIndex={-1} {...elemProps}>
-        {children}
-      </MenuList>
-    </Menu>
-  );
+  const {
+    buttonRef,
+    children,
+    error,
+    isAnimated,
+    isAutoFlipped,
+    isAutoFocused,
+    isFlipped,
+    isHidden,
+    isHiding,
+    menuRef,
+    onCloseOnEscape,
+    ...elemProps
+  } = props;
+
+  const popupRef = React.useRef<HTMLDivElement>(null);
+
+  const [width, setWidth] = useState(0);
+
+  const handleWidthChange = () => {
+    if (buttonRef.current) {
+      const newMenuWidth = buttonRef.current.clientWidth + 2 * buttonBorderWidth;
+      setWidth(newMenuWidth);
+    }
+  };
+
+  useEffect(() => {
+    handleWidthChange();
+  }, [width]);
+
+  // TODO: Figure out a better way to handle width changes in the reference button.
+  // Seems like we should resize the menu when the reference button width changes, not
+  // necessarily when the window resizes. Resizing the menu on window resize addresses
+  // the case when `grow = true` and the user resizes the browser window while the
+  // menu is open, but doesn't address cases where the reference button size changes
+  // through other means.
+  useEffect(() => {
+    // Update menu width state on resize to ensure menu resizes as window resizes
+    window.addEventListener('resize', handleWidthChange);
+
+    return () => {
+      window.removeEventListener('resize', handleWidthChange);
+    };
+  }, []);
+
+  useCloseOnEscape(popupRef, () => {
+    if (onCloseOnEscape) {
+      onCloseOnEscape();
+    }
+  });
```

Comment:
A bit cleaner is:
```suggestion
  useCloseOnEscape(popupRef, () => onCloseOnEscape?.());
```

I think that is a worthwhile terseness. It uses the optional chaining feature to call a callback if the function is defined. Perhaps we should allow `useCloseOnEscape` and `useCloseOnOutsideClick` to have an optional callback? This would drop the requirement to define an empty callback each render.

Reply:
Fixed.